transformer_engine/pytorch/fp8.py

```diff
@@ -599,10 +595,22 @@ def _default_sf_compute(
     fp8_max: float,
     margin: int,
 ) -> torch.Tensor:
-    """Default function to convert amax to scaling factor."""
+    """Default function to convert amax to scaling factor.
+    Computing the scaling factor requires consideration of the following scenarios:
+    1. amax == 0:
+       No action is possible, set scale to the previous scale (or 1).
+    2. 0 < amax < tiny_amax
+       The amax is too tiny that the scale becomes infinite in FP32.
+       Set scale = FP32_max
+    3. tiny_amax <= amax < FP32_max:
+       Set scale = FP8_max (or scaled_max) / amax
+    4. When amax == inf or amax == nan:
+       No action is possible, set scale to the previous scale (or 1).
+    """
     sf = (fp8_max / amax) / (2 ** margin)
     sf = torch.where(amax > 0.0, sf, scale)
     sf = torch.where(torch.isfinite(amax), sf, scale)
+    sf = torch.where(torch.isfinite(sf), sf, torch.ones_like(sf) * torch.finfo(torch.float32).max)
```

Comment:
Nit: this implementation is somewhat different than the CUDA kernel.
```suggestion
    sf = torch.where(torch.isinf(sf), torch.full_like(sf, torch.finfo(torch.float32).max), sf)
```
Alternatively, we could us `isfinite` in the CUDA kernel instead of `isinf`. That might be slightly more robust in case we run into NaNs, but I don't expect it to have a big effect.